src/common/snippets/src/lowered/pass/insert_load_store.cpp

```diff
@@ -67,7 +67,9 @@ bool InsertLoadStore::insert_load(LinearIR& linear_ir, const LinearIR::constExpr
         const auto load = std::make_shared<op::Load>(data_node->output(0), get_count(data_expr->get_output_port_descriptor(0)));
         PortDescriptorUtils::set_port_descriptor_ptr(load->output(0), consumer_input.get_descriptor_ptr()->clone());
         const auto load_expr = linear_ir.create_expression(load, {output_connector});
-        linear_ir.insert(std::find(data_expr_it, linear_ir.cend(), consumer_expr), load_expr);
+        const auto insertion_pos = std::find(data_expr_it, linear_ir.cend(), consumer_expr);
+        OPENVINO_ASSERT(insertion_pos != linear_ir.cend(), "Consumer should be after data producer in Linear IR");
```

Comment:
It looks like we should think strategically and implement `find` helpers that will encapsulate these checks. Otherwise this custom-checks approach could lead to considerable amount of code duplication in future. Probably the easiest way is to create `linear_ir.find_before(expression_it, target_expression)` and a similar `linear_ir.find_after(...)`. 

An alternative approach is to perform all-graph validation passes and rely on graph validity afterwards. This validation requires two passes: we enumerate nodes first, and then check that sourse has lower index than any of the consumers for every PortConnector.

What do you think?

Reply:
I fully agree with the both points. Thank you very much for the ideas! I like it.
As for first suggestion, I added these methods.
As for second, I've created the task 113536 for expression numeration. I hope we will implement it in the near future.